client/rest-high-level/src/main/java/org/elasticsearch/client/asyncsearch/GetAsyncSearchRequest.java

```diff
@@ -0,0 +1,76 @@
+/*
+ * Licensed to Elasticsearch under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+
+package org.elasticsearch.client.asyncsearch;
+
+import org.elasticsearch.client.TimedRequest;
+import org.elasticsearch.common.unit.TimeValue;
+
+import java.util.Objects;
+
+public class GetAsyncSearchRequest extends TimedRequest {
+
+    private TimeValue waitForCompletion;
```

Comment:
you need a default value here ?

Reply:
The defaults are set on the receiving Server side if we don't send these parameters over Rest, so if the user doesn't set them in the client they should be set by the RestGetAsyncSearchAction when parsing the request (https://github.com/elastic/elasticsearch/blob/master/x-pack/plugin/async-search/src/main/java/org/elasticsearch/xpack/search/RestGetAsyncSearchAction.java#L38). I think we usually try to not send default values over REST if possible thats why I left it here.

Reply:
Ok, thanks for explaining